### PR TITLE
SearchResultsWidget shows loader for "results count"

### PR DESCRIPTION
**Pull request by apepper**
For that the loading of "search.count()" had to move to "TotalCountSummary"

Needs content: https://edit.scrivito.com/d0a154d76edf2a7bd991fc658e700a1d~https://more-loading.scrivito-portal-app.pages.dev/en/search-results?q=Foo&_scrivito_workspace_id=k05aba3b533fa47f&_scrivito_display_mode=view

### CODE REVIEW

**Review by dcsaszar — APPROVED** [review]
Technically OK but design wise I'd prefer a different option.

Currently the spinner is hard to see on the background:

image


My favorite option:
* Some title, e.g., "Search results", is shown right from the start. (`loading: () => "Search results"`) This gives the user quick success feedback.
* The spinner is shown where the result list will appear. This moves attention to the place the user will look first anyway, and it looks good on the white background.

**Comment by apepper** [comment]
REVIEW ([2899b18](https://github.com/Scrivito/scrivito-portal-app/pull/403/commits/2899b186470411158ea0a69afad4c177f590df36))

Needs content: https://edit.scrivito.com/d0a154d76edf2a7bd991fc658e700a1d~https://more-loading.scrivito-portal-app.pages.dev/en/search-results?q=Foo&_scrivito_workspace_id=k05aba3b533fa47f&_scrivito_display_mode=view

Single commits are best viewed without whitespace changes.

File: src/Widgets/SearchResultsWidget/SearchResultsWidgetComponent.tsx [hunk 1]
```diff
@@ -113,9 +142,18 @@ provideComponent(SearchResultsWidget, ({ widget }) => {
...
       </section>
-    </InPlaceEditingOff>
-  )
-})
+    )
+  },
+  {
+    loading: () => (
+      <section className="bg-white py-3">
+        <div className="container text-center">
```
Comment (dcsaszar) [inline src/Widgets/SearchResultsWidget/SearchResultsWidgetComponent.tsx:150]: * [ ] FYI The code duplicated in the component and the loading could be moved to the parent component.
* [ ] FYI If we drop the `text-center` the loading becomes trivial. Maybe good enough?

File: src/Widgets/SearchResultsWidget/SearchResultsWidgetComponent.tsx [hunk 2]
```diff
@@ -138,5 +176,9 @@ const TotalCountSummary = connect(
       />
     )
   },
-  { loading: Loading },
+  {
+    loading: ({ widget }: { widget: SearchResultsWidgetInstance }) => (
+      <ContentTag content={widget} attribute="resultsLoadingHeadline" />
```
Comment (dcsaszar) [inline src/Widgets/SearchResultsWidget/SearchResultsWidgetComponent.tsx:181]: * [ ] FYI it turns out the `tag="span"` in the main component can be dropped.

**Review by dcsaszar — APPROVED** [review]
I like it. Some suggestions in comments.

File: src/Widgets/SearchResultsWidget/SearchResultsWidgetEditingConfig.ts [hunk 3]
```diff
@@ -7,6 +7,7 @@ provideEditingConfig(SearchResultsWidget, {
   thumbnail: Thumbnail,
   attributes: {
     readMoreLabel: { title: 'Read-more buttons label' },
+    resultsLoadingHeadline: { title: 'Headline for loading results' },
```
[the remark this answers is not in the document]
Comment (apepper) [reply src/Widgets/SearchResultsWidget/SearchResultsWidgetEditingConfig.ts:10]: I switched to `Headline while loading results`.

File: src/Widgets/SearchResultsWidget/SearchResultsWidgetComponent.tsx (hunk 1 above)
Comment (apepper) [reply src/Widgets/SearchResultsWidget/SearchResultsWidgetComponent.tsx]: I would like to keep the `text-center`. Otherwise the loader looks a bit lost.

**Pull request merged**